recipes/README.md

```diff
@@ -111,3 +111,101 @@ python3 convert.py /tmp/llama2/meta_model_0.pt --ctx 4096
 ```
 
 This will output a gguf file in the same precision which can be used for running inference.
+
+### Architecture Optimization
+
+TorchTune integrates with `torchao`(https://github.com/pytorch-labs/ao/) for architecture optimization techniques including quantization and sparsity. Currently only some quantization techniques are integrated, see `receipes/configs/quantize.yaml` for more details.
+
+#### Quantize
+To quantize a model (default is int4 weight only quantization):
+```
+tune run quantize --config quantize
+```
+
+#### Eval
+Evaluate a quantized model, add the following to `eleuther_eval.yaml`:
+```
+# Quantization specific args
+quantizer:
+  _component_: torchtune.utils.Int4WeightOnlyQuantizer
+  groupsize: 256
+```
+
+and run the eval command:
+```
+tune run eleuther_eval --config eleuther_eval
+```
+
+#### Generate
+Changes in generate.yaml
+```
+# Model arguments
+checkpointer:
+  checkpoint_files: [meta_model_0.8da4w.pt]
+
+# Quantization Arguments
+quantizer:
+  _component_: torchtune.utils.Int8DynActInt4WeightQuantizer
+  groupsize: 128
+```
+
+Changes in generate.py, comment out kv_cache, there is some issues now, we will fix later
```

Comment:
Sorry if this was discussed and I missed this. But whats the problem with KV Cache? Without this our generation will be really slow?